### PR TITLE
Enable filtering the Billed field on Cohorts and Cohort Samples views

### PR DESCRIPTION
**Pull request by qu8n**
Filtering on Cohorts view
Filtering on Cohort Samples view

### CODE REVIEW

File: frontend/src/components/RecordsList.tsx
Comment (qu8n) [inline frontend/src/components/RecordsList.tsx:1]: (For starter, I'm restricting my PR comments only to codes written outside of our pairing yesterday, but I'm happy to clarify anything here.)

Passed a new variable into our `prepareCohortDataForAgGrid()` function. This variable contains the filter option selected by users (examples below). We will take what user selected and manually filter the data inside `prepareCohortDataForAgGrid()` before feeding it into the AG Grid table component.
 
Examples of what `filterModel` returns:
```ts
// When both 'true' and 'false' are selected, it returns an empty object
{}

// When just 'Yes' is selected
{
    "billed": {
        "values": ["Yes"],
        "filterType": "set"
    }
}

// Nothing selected
{
    "billed": {
        "values": [],
        "filterType": "set"
    }
}
```

File: frontend/src/shared/helpers.tsx [hunk 1]
```diff
@@ -512,6 +515,83 @@ function setupEditableSampleFields(samplesColDefs: ColDef[]) {
...
+    if (selectedNone) {
+      newCohorts = [];
+      newCohortsConnection.totalCount = 0;
+    } else {
+      newCohorts = newCohorts.filter(
+        (cohort) => cohort.billed === selectedValues[0]
+      );
+      newCohortsConnection.totalCount = newCohorts.length;
+    }
+  }
```
Comment (qu8n) [inline frontend/src/shared/helpers.tsx:587]: Filter data based on what users want to filter by. Also calculate the new number of cohorts based on newly filtered data.

Reference the examples [here](https://github.com/mskcc/smile-dashboard/pull/120/files#r1581433690) on what `filterModel` looks like.

File: frontend/src/components/RecordsList.tsx
Comment (qu8n) [inline frontend/src/components/RecordsList.tsx:1]: (For starter, I'm restricting my PR comments only to codes written outside of our pairing yesterday, but I'm happy to clarify anything here.)

Passed a new variable into our `prepareCohortDataForAgGrid()` function. This variable contains the filter option selected by users (examples below). We will take what user selected and manually filter the data inside `prepareCohortDataForAgGrid()` before feeding it into the AG Grid table component.
 
Examples of what `filterModel` returns:
```ts
// When both 'Yes' and 'No' are selected, it returns an empty object
{}

// When just 'Yes' is selected
{
    "billed": {
        "values": ["Yes"],
        "filterType": "set"
    }
}

// Nothing selected
{
    "billed": {
        "values": [],
        "filterType": "set"
    }
}
```

File: frontend/src/shared/helpers.tsx (hunk 1 above)
Comment (qu8n) [inline frontend/src/shared/helpers.tsx:587]: Filter data based on what users want to filter by. Also calculate the new number of cohorts based on newly filtered data.

Reference the examples [here](https://github.com/mskcc/smile-dashboard/pull/120/files#r1581458274) on what `filterModel` looks like.

**Review by ao508 — APPROVED** [review]

**Pull request merged**